tests/unit/test_num.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Redis Ltd. 2018 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "src/util/num.h"
+
+#include "acutest.h"
+
+#define TYPE(width) uint##width ## _t
+#define FUNC(width) checked_add_u ## width
+
+#define DEFINE_TEST(name, width) \
+void test_NumOverflow_CheckedAddU ##width (void) { \
+    TYPE(width) out = 0; \
+    static const TYPE(width) MAX = UINT##width##_MAX; \
+    bool is_ok = FUNC(width)(MAX - 1, 1, &out); \
```

Comment:
Please add a short comment, e.g. Adding MAX - 1 + 1 should return MAX, no overflow